server/datastore/mysql/migrations/tables/20220818101352_ChangeSoftwareVendorWidth.go

```diff
@@ -38,6 +32,14 @@ func Up_20220818101352(tx *sql.Tx) error {
 		return errors.Wrapf(err, "updating temp vendor column")
 	}
 
+	//---------------------
+	// Add uniq constraint
+	//---------------------
+	if _, err := tx.Exec(
+		"ALTER TABLE software ADD constraint unq_name UNIQUE (name, version, source, `release`, vendor_wide, arch)"); err != nil {
+		return errors.Wrapf(err, "adding new uniquess constraint")
+	}
+
```

Comment:
So if you set the vendor_wide before doing this, you will dodge the case where there are two rows that are the same except for vendor? That sounds like such a weird case! ha

Reply:
Yup - `vendor_wide` defaults to `''` which was kind of the root of the problem but, if we run the update before adding the new constrain then there shouldn't be any issues because the old constrain guarantees the uniqueness of  (name, version, source, release, vendor, arch).

Yeah kind of a weird edge case lol